### PR TITLE
[#3211] Xenial support

### LINKED ISSUE #3211: Add Ubuntu 16.04 LTS (Xenial) support to install scripts
Released today https://insights.ubuntu.com/2016/04/20/canonical-unveils-6th-lts-release-of-ubuntu-with-16-04/

Related: https://github.com/mysociety/alaveteli/issues/2019

### PR DESCRIPTION
**Pull request by gbp**
## Relevant issue(s)

Fixes #3211 

## What does this do?

Adds support for Xenial and switches Travis over to use it instead on Trusty.

## Why was this needed?

We need to support a newer Ubuntu as Trusty is EOL and we need to drop support see: #4970.

## Implementation notes

My `vagrant up` & `vagrant provision` commands took a very long time to run this turned out to be an Vagrant issue with the shared folders. I doubt this is due to Xenial, more probably due to my setup, when I switched to using `rsync` everything was fast.

### CODE REVIEW

**Comment by garethrees** [comment]
> My `vagrant up` & `vagrant provision` commands took a very long time to run this turned out to be an Vagrant issue with the shared folders. I doubt this is due to Xenial, more probably due to my setup, when I switched to using `rsync` everything was fast.

Could be similar to https://github.com/mysociety/alaveteli/pull/4218#issuecomment-329799546

**Comment by gbp** [comment]
Setting `BUNDLE_PATH` in `config/general.yml` to be outside the vagrant shared folder EG: `../alaveteli-bundle` seems to make everything fast.

**Comment by garethrees** [comment]
Hmm that should get set to `~/bundle` https://github.com/mysociety/alaveteli/blob/develop/script/install-as-user#L114

**Comment by gbp** [comment]
Seems to be a Postfix issue when the hostname isn't set, this should fix but unsure if there are other considerations here. If possible we should run for all OSs so we'll want to check this change first.

```diff
diff --git a/script/site-specific-install.sh b/script/site-specific-install.sh
index a0bd298fa..951da6741 100755
--- a/script/site-specific-install.sh
+++ b/script/site-specific-install.sh
@@ -74,6 +74,12 @@ then
   apt-get install -y ruby2.1 ruby2.1-dev
 fi
 
+# Ubuntu Xenial Fixes
+if [ x"$DISTRIBUTION" = x"ubuntu" ] && [ x"$DISTVERSION" = x"xenial" ]
+then
+  hostnamectl set-hostname $HOST
+fi
+
 update_mysociety_apt_sources
 
 apt-get -y update
```

**Comment by gbp** [comment]
> Hmm that should get set to `~/bundle` https://github.com/mysociety/alaveteli/blob/develop/script/install-as-user#L114

Ah that'll be because I'm recycling my old `general.yml` which was generated from the example file.

**Comment by garethrees** [comment]
I get errors on a clean install:

```
cd /tmp/
git clone https://github.com/mysociety/alaveteli.git
cd alaveteli
git submodule update --init
git co -b 3211-xenial-support origin/3211-xenial-support

# Edit .vagrant.yml…
cat .vagrant.yml
fqdn: 10.10.10.31
ip: 10.10.10.31
os: xenial64
memory: 4096
cpus: 2
themes_dir: '/Users/gareth/Code/alaveteli-themes'

v up

snip…

    default: Reading package lists...
    default: Setting hostname...
    default: Installing postfix...
    default: E
    default: :
    default: Sub-process /usr/bin/dpkg returned an error code (1)
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.
```

**Comment by gbp** [comment]
> I get errors on a clean install:

@garethrees this looks like the error which was fixed with f4a5313, did you run this recently?

**Comment by gbp** [comment]
@garethrees ah, in your `.vagrant.yml` your FQDN is just an IP address, this isn't a valid domain name as as such Postfix isn't happy. Is this how you normally configure Vagrant? We came across a similar issue here https://github.com/mysociety/alaveteli/pull/4522

**Comment by gbp** [comment]
Here is the error being thrown:
```
    default: postconf: warning: valid_hostname: numeric hostname: 10.10.10.31
    default: postconf: fatal: unable to use my own hostname
    default: postconf: warning: valid_hostname: numeric hostname: 10.10.10.31
    default: postconf: fatal: unable to use my own hostname
```

**Comment by garethrees** [comment]
> your FQDN is just an IP address, this isn't a valid domain name as as such Postfix isn't happy.

Yeah. Changed it to use nip.io and it works okay.

**Comment by gbp** [comment]
> Worth checking that hostnamectl works as expected on the other systems

Yes. I've done this already. They all work fine.

File: lib/tasks/themes.rake [hunk 1]
```diff
@@ -49,7 +49,7 @@ namespace :themes do
 
   def move_old_theme(old_theme_directory)
     puts "There was an old-style theme at #{old_theme_directory}" if verbose
-    moved_directory = "#{old_theme_directory}-moved"
+    moved_directory = "#{old_theme_directory}-moved".gsub('/-', '-')
```
[the remark this answers is not in the document]
Comment (lizconlan) [reply lib/tasks/themes.rake]: That's clearer, have pushed a fixup!

File: Vagrantfile [hunk 2]
```diff
@@ -86,7 +86,7 @@ DEFAULTS = {
   'ip' => '10.10.10.30',
   'memory' => 1536,
   'themes_dir' => '../alaveteli-themes',
-  'os' => 'jessie64',
+  'os' => 'xenial64',
```
Comment (garethrees) [inline Vagrantfile]: I'd missed this initially. This change is smuggled in – the commit is about adding Xenial support, not about changing defaults etc.

I've made a PR to update the default at https://github.com/mysociety/alaveteli/pull/5026, so lets drop this line from this PR.

**Pull request merged**